pkg/apis/agones/v1/gameserver.go

```diff
@@ -926,3 +926,95 @@ func (gs *GameServer) Patch(delta *GameServer) ([]byte, error) {
 	result, err = json.Marshal(patch)
 	return result, errors.Wrapf(err, "error creating json for patch for GameServer %s", gs.ObjectMeta.Name)
 }
+
+// UpdateCount increments or decrements a CounterStatus on a Game Server by the given amount.
+func (gs *GameServer) UpdateCount(name string, action string, amount int64) error {
+	err := fmt.Errorf("unable to UpdateCount: Name %s, Action %s, Amount %d", name, action, amount)
+	if !(action == GameServerAllocationIncrement || action == GameServerAllocationDecrement) || amount < 0 {
+		return err
+	}
+	if counter, ok := gs.Status.Counters[name]; ok {
+		cnt := counter.Count
+		if action == GameServerAllocationIncrement {
+			cnt += amount
+			// only check for Count > Capacity when incrementing
+			if cnt > counter.Capacity {
+				return err
+			}
+		}
+		if action == GameServerAllocationDecrement {
+			cnt -= amount
+		}
+		if cnt < 0 {
```

Comment:
Just had a thought -- do we want to only check for Count < 0 when decrementing?

Reply:
Agreed, I think that's good enough.